Cython/Compiler/Symtab.py

```diff
@@ -919,17 +920,33 @@ def find_imported_module(self, path, pos):
                 return None
         return scope
 
-    def lookup(self, name):
+    def lookup(self, name, _no_mangle=False):
```

Comment:
Hmmm … not … beautiful. Can't we handle this with scope recursion somehow, instead of letting the Scope base class know how and when to mangle Python names or not? I mean, it's ok if a class scope knows that it can look up a name in different ways. Other scopes shouldn't have to care.

Reply:
I think if we handled with scope recursion then:

```
cdef int __var = 1
cdef int _A__var = 2

class A:
  class B:
    def get_var(self):
      return __var
```

would end up returning `_A__var` instead of `__var` because after `B` fails to find the name it'd then let `A` have first go.

Maybe that's behaviour's OK? It isn't obvious to me either way.

I'll have a look and see if it can be done better, but the challenge is we can have multiple classes with an opinion about mangling in the chain. In Python only the "current" class gets to decided, and my inclination says it should be the "current" class or no mangling at all.

Reply:
Is `A` even in the `parent_scope` chain of `B`? AFAIK, that's not how class scopes work in Python. They do not nest.

Reply:
You're right. This can be (and now is) done much cleaner